### PR TITLE
Update LRG selection cuts for SV2

### PR DESCRIPTION
**Pull request by rongpu**
Changes from the previous "main" LRG cuts:
 - Switch from optical to IR selection
 - Tweak the low-z and and sliding cuts
 - Remove bright (G<18) GAIA sources

### CODE REVIEW

**Comment by geordie666** [comment]
I think this looks OK, but please don't forget to update the `doc/changes.rst` file before you merge.

**Comment by geordie666** [comment]
Oh, and also please update [the SV2 wiki](https://desi.lbl.gov/trac/wiki/TargetSelectionWG/SV2).

**Comment by geordie666** [comment]
Oh, and it looks like unit tests failed first time round so nobody should merge this yet.

**Comment by rongpu** [comment]
Thanks @geordie666 for the reminder. I have updated both `doc/changes.rst` and the SV2 wiki.

File: py/desitarget/sv2/sv2_cuts.py [hunk 1]
```diff
@@ -313,7 +313,8 @@ def isLRG(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
     # ADM color-based selection of LRGs.
     lrg &= isLRG_colors(
         gflux=gflux, rflux=rflux, zflux=zflux, w1flux=w1flux,
-        zfiberflux=zfiberflux, south=south, primary=primary
+        zfiberflux=zfiberflux, gaiagmag=gaiagmag, south=south,
```
Comment (geordie666) [inline py/desitarget/sv2/sv2_cuts.py:316]: The `gaiagmag=gaiagmag` here, needs to instead be in the call to `notinLRG_mask()` (about 10 lines further up), as that is the function in which you actually apply the limits on Gaia G.

File: py/desitarget/sv2/sv2_cuts.py [hunk 2]
```diff
@@ -277,8 +277,8 @@ def isBACKUP(ra=None, dec=None, gaiagmag=None, primary=None):
 
 def isLRG(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
           zfiberflux=None, rfluxivar=None, zfluxivar=None, w1fluxivar=None,
-          gnobs=None, rnobs=None, znobs=None, maskbits=None, primary=None,
-          south=True):
+          gaiagmag=None, gnobs=None, rnobs=None, znobs=None, maskbits=None,
```
Comment (geordie666) [inline py/desitarget/sv2/sv2_cuts.py:280]: Here, you have correctly added `gaiagmag=None`. But, you also need to pass `gaiagmag` to the `isLRG()` function in the first place. 

Down around line 1860 you'll see this code:
```
     if "LRG" in tcnames:
         for south in south_cuts:
             lrg_classes[int(south)] = isLRG(
```
you need to also make sure to pass `gaiagmag=gaiagmag` to the `isLRG` function at that point in the code.

**Review by geordie666 — COMMENTED** [review]
@rongpu: You've forgotten to pass the `gaiagmag` variable in a couple of places (or passed it to the wrong function). If you make my suggested changes, you'll know that everything is correct because unit tests will pass.

I'll check back in an hour or so, and if the changes look correct I'll merge this PR.

**Comment by rongpu** [comment]
Ugh, these are very silly mistakes. I have fix them (hopefully).

**Comment by geordie666** [comment]
Thanks, @rongpu, this looks great now. I'll merge soon.

**Comment by rongpu** [comment]
Thanks!

**Pull request merged**